OpenEdXMobile/src/main/java/org/edx/mobile/view/TabsBaseFragment.java

```diff
@@ -42,6 +45,28 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         return binding.getRoot();
     }
 
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        // Handle tab-selection through deep link (if available)
+        if (getArguments() != null && binding != null) {
+            @ScreenDef final String screenName = getArguments().getString(Router.EXTRA_SCREEN_NAME);
+            if (screenName != null) {
```

Comment:
This condition could be optimized in this way
`if (screenName != null && screenName.equals(Screen.COURSE_VIDEOS)) {}`
but if its good for future use when we shall need to implement deep linking on other tabs.
Thoughts ?

Reply:
This won't be an optimization rather hard coding the code-block to only work for COURSE_VIDEOS screen. The current implementation is done to make the selection of any tab 
 easy in the future by only adding an `else if` clause in the for-loop.